sui/its-example.js

```diff
@@ -0,0 +1,499 @@
+        arguments: [ItsToken.objects.TreasuryCap, unitAmount],
+        typeArguments: [ItsToken.typeArgument],
+    });
+
+    await txBuilder.moveCall({
+        target: `${Example.address}::its::send_interchain_transfer_call`,
+        arguments: [
+            objectIds.singleton,
+            objectIds.its,
+            objectIds.gateway,
+            objectIds.gasService,
+            TokenId,
+            Coin,
+            destinationChain,
+            destinationAddress,
+            '0x', // its token metadata
+            walletAddress,
+            gas,
+            '0x', // gas params
+            CLOCK_PACKAGE_ID,
+        ],
+        typeArguments: [ItsToken.typeArgument],
+    });
+
+    await broadcastFromTxBuilder(txBuilder, keypair, `${amount} ${symbol} Token Sent`);
+}
+
+async function receiveToken(keypair, client, contracts, args, options) {
+    const itsData = options.data;
+    const { Example, RelayerDiscovery, AxelarGateway, ITS } = contracts;
+    const [sourceChain, messageId, sourceAddress, tokenSymbol, amount] = args;
+
+    checkTrustedAddresses(ITS.trustedAddresses, sourceChain, sourceAddress);
+
+    // Prepare Object Ids
+    const symbol = tokenSymbol.toUpperCase();
+    const ids = {
+        discovery: RelayerDiscovery.objects.RelayerDiscoveryv0,
+        gateway: AxelarGateway.objects.Gateway,
+        itsChannel: ITS.objects.ChannelId,
+        exampleChannel: options.channelId || Example.objects.ItsChannelId,
+    };
+
+    if (!ids.exampleChannel) {
+        throw new Error('Please provide either a channel id (--channelId) or deploy the Example contract');
+    }
+
+    if (!contracts[symbol]) {
+        throw new Error(`Token ${symbol} not found. Deploy it first with 'node sui/its-example.js deploy-token' command`);
+    }
+
+    const Token = contracts[symbol];
+    const unitAmount = getUnitAmount(amount, Token.decimals);
+    const tokenId = Token.objects.TokenId;
+
+    const payload = defaultAbiCoder.encode(
+        ['uint256', 'uint256', 'bytes', 'bytes', 'uint256', 'bytes'],
+        [ITSMessageType.InterchainTokenTransfer, tokenId, sourceAddress, ids.exampleChannel, unitAmount, itsData],
+    );
+
+    // Get Discovery table id from discovery object
+    const transactionList = await getTransactionList(client, ids.discovery);
+
+    // Find the transaction with associated channel id
+    const transaction = transactionList.find((row) => row.name.value === ids.exampleChannel);
+
+    const receiveTxBuilder = new TxBuilder(client);
+
+    // Take the approved message from the gateway contract.
+    const approvedMessage = await receiveTxBuilder.moveCall({
+        target: `${AxelarGateway.address}::gateway::take_approved_message`,
+        arguments: [ids.gateway, sourceChain, messageId, sourceAddress, ids.itsChannel, payload],
+    });
+
+    const { moduleName, name, packageId, txArgs } = await parseExecuteDataFromTransaction(client, transaction, approvedMessage);
```

Comment:
You need to recursively find the call you need to make, check out https://github.com/axelarnetwork/axelar-cgp-sui/blob/main/test/testutils.js#L268.

Reply:
Check out https://github.com/axelarnetwork/axelar-cgp-sui/pull/187 which converts the cgp-sui examples you wrote to using relayer discovery.